fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexerByIndex.java

```diff
@@ -86,13 +86,13 @@
 
                     if (common.isSyntheticIndex() ||
                             common.recordTypes.size() != 1) {
-                        buildIndexFromIndexThrowEx("IndexFromIndex: tgt cannot be scanned from index");
+                        buildIndexFromIndexThrowEx("IndexFromIndex: this index cannot be scanned from another index");
```

Comment:
These error messages are somewhat opaque because all of the different clauses lead to the same message. It would be nice if they included a reason (e.g., "this index cannot be be built by scanning another index as the target index has too many types").

I'm also not sure the `IndexFromIndex` prefix helps. It may make sense to instead have it be its own exception class instead (`BuildIndexFromIndexValidationException` or something)

Reply:
Maybe `BuildIndexFromIndex.ValidationException`?